### ISSUE TITLE
FFXIVLauncher

### ISSUE THREAD

**Issue by traxys** [NixOS/nixpkgs#172750]
**Project description**
Launcher for Final Fantasy XIV

There is a Arch Linux PKGBUILD for example [here](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=xivlauncher)

**First steps**

```nix
pkgs.buildDotnetModule rec {
    pname = "FFXIVQuickLauncher";
    version = "6.2.31";

    src = pkgs.fetchgit {
        url = "https://github.com/goatcorp/FFXIVQuickLauncher";
        rev = version;
        sha256 = "sha256-TZAn0AC6ovGylHLGE7EO9BCjbyDEBk7n/8+XaZzr27A=";
        leaveDotGit = true;
     };

    projectFile = "src/XIVLauncher.Core/XIVLauncher.Core.csproj";

    dotnetFlags = [ "--runtime linux-x64" ];
    #dotnetBuildFlags = [ "--sc" ];

    nativeBuildInputs = [ pkgs.git ];
    runtimeDeps = [ pkgs.SDL2 ];

    dotnet-sdk = pkgs.dotnetCorePackages.sdk_6_0;
    dotnet-runtime = pkgs.dotnetCorePackages.runtime_6_0;

    nugetDeps = ./xivdeps.nix;
 }
```

This fails at runtime with 

```
A fatal error was encountered. The library 'libhostpolicy.so' required to execute the application was not found in '/nix/store/yp2hd12wap9pz99yyhy3fybwm745ahhw-dotnet-runtime-6.0.4'.
Failed to run as a self-contained app.
  - The application was run as a self-contained app because '/nix/store/bdv1832m7wyd0af1169pxxnfw3pwn6nf-FFXIVQuickLauncher-6.2.31/lib/FFXIVQuickLauncher/XIVLauncher.Core.runtimeconfig.json' did not specify a framework.
  - If this should be a framework-dependent app, specify the appropriate framework in '/nix/store/bdv1832m7wyd0af1169pxxnfw3pwn6nf-FFXIVQuickLauncher-6.2.31/lib/FFXIVQuickLauncher/XIVLauncher.Core.runtimeconfig.json'.
  ```

And I don't have enough CSharp experience to understand this.

To generate `xivdeps.nix` you can download the source code of XIVLauncher, go into `src/XIVLauncher.Core/` and run `dotnet restore -r linux-x64 XIVLauncher.Core.csproj --packages ./nuget_tmp.packages`, followed by `nuget-to-nix ./nuget_tmp.packages > xivdeps.nix`. You can get both dependencies with `nix shell nixpkgs#dotnet-sdk nixpkgs#nuget-to-nix`.

**Metadata**

* homepage URL: https://goatcorp.github.io/
* source URL: https://github.com/goatcorp/FFXIVQuickLauncher
* license: GPLv3
* platforms: linux

**Comment by LunNova** [comment]
Fixed by #169121

**Comment by IvarWithoutBones** [comment]
> To generate xivdeps.nix you can download the source code of XIVLauncher, go into src/XIVLauncher.Core/ and run dotnet restore -r linux-x64 XIVLauncher.Core.csproj --packages ./nuget_tmp.packages, followed by nuget-to-nix ./nuget_tmp.packages > xivdeps.nix. You can get both dependencies with nix shell nixpkgs#dotnet-sdk nixpkgs#nuget-to-nix.

A passthru script to generate the update script is actually provided by `buildDotnetModule`, you can build it with `nix build -L nixpkgs#xivlauncher.fetch-deps`. Running the `result` script that generates should push the generated lockfile to `/tmp`.
